packages/contracts-bedrock/contracts/libraries/SafeCall.sol

```diff
@@ -82,13 +111,11 @@ library SafeCall {
                 revert(28, 100)
             }
 
-            // The call will be supplied at least (((_minGas + 200) * 64) / 63) - 49 gas due to the
-            // above assertion. This ensures that, in all circumstances, the call will
-            // receive at least the minimum amount of gas specified.
-            // We can prove this property by solving the inequalities:
-            // ((((_minGas + 200) * 64) / 63) - 49) >= _minGas
-            // ((((_minGas + 200) * 64) / 63) - 51) * (63 / 64) >= _minGas
-            // Both inequalities hold true for all possible values of `_minGas`.
+            // The call will be supplied at least ((_minGas * 64) / 63) gas due to the
+            // above assertion. This ensures that, in all circumstances (except for when the
+            // `_minGas` does not account for the `memory_expansion_cost` and `code_execution_cost`
+            // factors of the dynamic cost of the `CALL` opcode), the call will receive at least
+            // the minimum amount of gas specified.
             _success := call(
                 gas(), // gas
```

Comment:
Comment only:

It's interesting to me that we give `gas()` (or rather `min(gas() * 63 / 64, gas())`) to the actual `CALL`, rather than `_minGas`, which in the Portal is `tx.gasLimit`.

https://github.com/ethereum-optimism/optimism/blob/0470651b3c425a74f9ba24113f0a695645e8a7f9/packages/contracts-bedrock/contracts/L1/OptimismPortal.sol#L394

It seems fine since we do all the work above to ensure that when we get here, `gas()` is going to be plenty higher `tx.gasLimit`. 

Also, for whatever reason `CALL` doesn't seem to fail if `gas()` is less then the stack value consumed by gas.

Reply:
I think it makes sense to pass any additional gas a caller provides into the subcontext since the parameter is specifically annotated as a _minimum_ gas limit.

Reply:
Yeah, I agree. Just needed to think it through.